server/observable.go

```diff
@@ -504,12 +527,50 @@ func (o *Observable) updateStore() {
 	}
 }
 
+func (o *Observable) shouldSample() bool {
+	if o.sampleFrequency <= 0 {
+		return false
+	}
+
+	if o.sampleFrequency >= 100 {
+		return true
+	}
+
+	if mrand.Int31n(100) <= o.sampleFrequency) {
+		return true
+	}
+
+	return false
+}
+
+func (o *Observable) sampleAck(sseq, dseq uint64) {
+	if !o.shouldSample() {
+		return
+	}
+
+	e := &ObservableSampleEvent{
+		MsgSet:     o.mset.Name(),
+		Observable: o.name,
+		ObsSeq:     dseq,
+		Delay:      int64(time.Now().UnixNano()) - o.pending[sseq],
+	}
+
+	j, err := json.MarshalIndent(e, "", "  ")
```

Comment:
not really a fan of json marshaling here, its under lock and will slow things down, we should consider making the jsPubMsg accept an `interface{}` and do the marshalling in that go routine if we want to do this kind of thing

Reply:
Should not hold a lock for sure. I have been exploring options to make these faster.

Reply:
will leave this till there's a better option then